documentation/src/main/asciidoc/ch09.asciidoc

```diff
@@ -370,6 +370,100 @@ include::{sourcedir}/org/hibernate/validator/referenceguide/chapter09/Bootstrapp
 Refer to <<section-fail-fast>> and <<section-programmatic-api>> to learn more about the fail fast
 mode and the constraint declaration API.
 
+[[section-script-evaluator-factory]]
+==== Configuring script evaluator factory (`ScriptEvaluatorFactory`)
+
+For constrains like `@ScriptAssert` and `@ParameterScriptAssert` it might be useful to configure
+how scripts are evaluated, and which scripting engines are used. This can be done by setting
+a custom implementation of `ScriptEvaluatorFactory`. In particular this is important for
+OSGi environments (or similar to it) where user might face issues with dynamic class loading
+and https://www.jcp.org/en/jsr/detail?id=223[JSR-223]. It also would allow to use any custom
+script evaluation, not necessarily based on the https://www.jcp.org/en/jsr/detail?id=223[JSR-223].
+
+===== Programmatic configuration
+There are two ways to change the default `ScriptEvaluatorFactory` programmatically.
+First one is to use `hibernate.validator.script_evaluator_factory` property as described
+in <<section-provider-specific-settings>> section, by passing a fully qualified class name of `ScriptEvaluatorFactory`
+implementation to be used. In this case configuration will look like:
+
+[[example-script-evaluator-factory-via-addproperty]]
+.Changing `ScriptEvaluatorFactory` via `addProperty()`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter09/BootstrappingTest.java[tags=scriptEvaluatorFactoryViaAddProperty]
+----
+====
+
+Another option is to pass an instance of `ScriptEvaluatorFactory` when building `ValidatorFactory`.
+This gives more flexibility in configuration of the `ScriptEvaluatorFactory`. <<example-script-evaluator-factory-programmatically>>
+shows how this can be done.
+
+[[example-script-evaluator-factory-programmatically]]
+.Changing `ScriptEvaluatorFactory` by configuring `ValidatorFactory`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter09/BootstrappingTest.java[tags=scriptEvaluatorFactoryProgrammatically]
+----
+====
+
+===== XML configuration
+To specify `ScriptEvaluatorFactory` via XML the same `hibernate.validator.script_evaluator_factory`
+property can be used.
+
+[[example-script-evaluator-factory-xml]]
+.Method constraints configured via XML
+====
+[source, XML, indent=0]
+----
+include::{resourcesdir}/org/hibernate/validator/referenceguide/chapter09/script-evaluator-factory-validation.xml[]
+----
+====
+
+In case of using property to configure a factory (either in XML configuration, or by adding it programmatically)
+user is limited to use factories that has constructors without parameters.
+
+===== Predefined `ScriptEvaluatorFactory` implementations
+Hibernate Validator provides multiple implementations of `ScriptEvaluatorFactory` that user
```

Comment:
OK, I have a couple more comments for the rest of the wording but it's probably easier if I fix it on merging.